lnwire/typed_fee.go

```diff
@@ -0,0 +1,61 @@
+package lnwire
+
+import (
+	"io"
+
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	FeeRecordType tlv.Type = 55555
+)
+
+// Fee represents a fee schedule.
+type Fee struct {
+	BaseFee int32
+	FeeRate int32
+}
+
+// Record returns a TLV record that can be used to encode/decode the fee
+// type from a given TLV stream.
+func (l *Fee) Record() tlv.Record {
+	return tlv.MakeStaticRecord(
+		FeeRecordType, l, 8, feeEncoder, feeDecoder,
+	)
+}
+
+// feeEncoder is a custom TLV encoder for the fee record.
+func feeEncoder(w io.Writer, val interface{}, buf *[8]byte) error {
+	if v, ok := val.(*Fee); ok {
+		if err := tlv.EUint32T(w, uint32(v.BaseFee), buf); err != nil {
```

Comment:
I don't think it causes any problem, because signed and unsigned ints are encoded identical. Inside the `tlv` package, `binary` is used that also only offers unsigned put functions.